### DISCUSSION TITLE
Make not successful getting - Error: 64bit mode not supported on `i386'

### DISCUSSION THREAD

**Discussion by forwardcomputers** [ipxe/ipxe#1121] [Q&A] [answered]
Hi,

Trying to compile ipxe on Debian trixie 12.2, and keep getting this error when trying to get 'ipxe.efi'.
Running 'make' on it's own works fine but doesn't get me the efi files.
Any help is appreciated.

Ali.

```
/ipxe/src# make bin-x86_64-efi/ipxe.efi
  [PARSEROM]
  [DEPS] core/version.c
  [DEPS] arch/x86/transitions/librm.S
  ...
  [BUILD] bin-x86_64-efi/setjmp.o
  [BUILD] bin-x86_64-efi/patch_cf.o
arch/x86/core/patch_cf.S: Assembler messages:
arch/x86/core/patch_cf.S:27: Error: 64bit mode not supported on `i386'.
make: *** [Makefile.housekeeping:964: bin-x86_64-efi/patch_cf.o] Error 1
make: *** Waiting for unfinished jobs....
```

```
/ipxe/src# uname -a
Linux update 6.2.16-19-pve #1 SMP PREEMPT_DYNAMIC PMX 6.2.16-19 (2023-10-24T12:07Z) x86_64 GNU/Linux
```

```
/ipxe/src# as --version
GNU assembler (GNU Binutils for Debian) 2.41.50.20231227
```

```
/ipxe/src# gcc --version
gcc (Debian 13.2.0-9) 13.2.0
```

**Comment by mcb30** [comment]
On 11/01/2024 23:25, Ali Mustakim wrote:
> Trying to compile ipxe on Debian trixie 12.2, and keep getting this 
> error when trying to get 'ipxe.efi'.
> Running 'make' on it's own works fine but doesn't get me the efi files.
> Any help is appreciated.

This looks likely to be the same issue as

   https://github.com/ipxe/wimboot/commit/47efa60

and the same fix (moving ".arch i386" so that it comes after a ".code32" 
or ".code16") should therefore work.

Could you give that a try and report back?

Thanks,

Michael

**Reply by forwardcomputers** [reply]
Thanks for the response Michael.

Moving ".arch i386" after ".code16/32" or adding ".code16" before fixed the errors.
There are a few other files where this errors out
```
arch/x86/core/stack.S
arch/x86/core/stack16.S
arch/x86/interface/pcbios/e820mangler.S
arch/x86/interface/pxe/pxe_entry.S
arch/x86/prefix/bootpart.S
arch/x86/prefix/dskprefix.S
arch/x86/prefix/exeprefix.S
arch/x86/prefix/hdprefix.S
arch/x86/prefix/pxeprefix.S
arch/x86/prefix/libprefix.S
arch/x86/prefix/lkrnprefix.S
arch/x86/prefix/mbr.S
arch/x86/prefix/nullprefix.S
arch/x86/prefix/rawprefix.S
arch/x86/prefix/unlzma.S
arch/x86/prefix/usbdisk.S
arch/x86/transitions/liba20.S
arch/x86/transitions/libkir.S

```

**Comment by mcb30** [comment, marked as answer]
Should be fixed as of commit https://github.com/ipxe/ipxe/commit/6ca597eee

**Reply by forwardcomputers** [reply]
Thank you for the update.
